src/node_file.cc

```diff
@@ -3149,14 +3149,9 @@ void BindingData::LegacyMainResolve(const FunctionCallbackInfo<Value>& args) {
     }
   }
 
-  std::optional<std::string> module_path =
-      node::url::FileURLToPath(env, *package_json_url);
+  std::string module_path = *initial_file_path + ".js";
```

Comment:
Is the `.js` correct here? If there was an `index.json`, or `index.node` that would also work

Reply:
Updated to change to use the `main` value when it's defined, otherwise, fallback to `index.js`.